server/data/data-file/src/main/java/org/apache/james/mailrepository/file/MBoxMailRepository.java

```diff
@@ -250,15 +250,14 @@ private MimeMessage convertTextToMimeMessage(String emailBody) {
     }
 
     /**
-     * Generate a hex representation of an MD5 checksum on the emailbody
+     * Generate a hex representation of a SHA-256 checksum on the email body
      * 
      * @param emailBody
      * @return A hex representation of the text
      * @throws NoSuchAlgorithmException
      */
     private String generateKeyValue(String emailBody) throws NoSuchAlgorithmException {
-        // MD5 the email body for a reilable (ha ha) key
-        byte[] digArray = MessageDigest.getInstance("MD5").digest(emailBody.getBytes());
+        byte[] digArray = MessageDigest.getInstance("SHA-256").digest(emailBody.getBytes());
```

Comment:
`Hashing.sha256().hashBytes(emailBody.getBytes())` avoid some problems

Reply:
:trollface:  you forgot to use `StandardCharsets.UTF8` in the  `getBytes` method.